components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go

```diff
@@ -0,0 +1,357 @@
+// getRequestsForAutomateAsSource gives the requests for all the ports and types automate as source
+func getRequestsForAutomateAsSource(config models.Config) []models.FirewallRequest {
+	var reqBodies []models.FirewallRequest
+	// Dest postgres
+	for _, sourceNodeIP := range config.Hardware.AutomateNodeIps {
+		for _, destNodeIP := range config.Hardware.PostgresqlNodeIps {
+			reqBody := models.FirewallRequest{
+				SourceNodeIP:               sourceNodeIP,
+				DestinationNodeIP:          destNodeIP,
+				DestinationServicePort:     "7432",
+				DestinationServiceProtocol: "tcp",
+			}
+			reqBodies = append(reqBodies, reqBody)
+		}
+
+		for _, destNodeIP := range config.Hardware.OpenSearchNodeIps {
+			reqBody := models.FirewallRequest{
+				SourceNodeIP:               sourceNodeIP,
+				DestinationNodeIP:          destNodeIP,
+				DestinationServicePort:     "9200",
+				DestinationServiceProtocol: "tcp",
+			}
+			reqBodies = append(reqBodies, reqBody)
+		}
+	}
+	return reqBodies
+
+}
+
+// getRequestsForChefServerAsSource gives the requests for all the ports where chefserver is the source
+func getRequestsForChefServerAsSource(config models.Config) []models.FirewallRequest {
+
+	var reqBodies []models.FirewallRequest
+	for _, sourceNodeIP := range config.Hardware.ChefInfraServerNodeIps {
+		for _, destNodeIP := range config.Hardware.AutomateNodeIps {
+			nodeCert := getCertForNodeFromConfig(config, destNodeIP)
+			reqBody := models.FirewallRequest{
+				SourceNodeIP:               sourceNodeIP,
+				DestinationNodeIP:          destNodeIP,
+				DestinationServicePort:     "443",
+				DestinationServiceProtocol: "https",
+				Cert:                       nodeCert.Cert,
+				Key:                        nodeCert.Key,
+				RootCert:                   config.Certificate.RootCert,
+			}
+			reqBodies = append(reqBodies, reqBody)
+		}
+
+		// Dest postgres
+		for _, destNodeIP := range config.Hardware.PostgresqlNodeIps {
+			reqBody := models.FirewallRequest{
+				SourceNodeIP:               sourceNodeIP,
+				DestinationNodeIP:          destNodeIP,
+				DestinationServicePort:     "7432",
+				DestinationServiceProtocol: "tcp",
+			}
+			reqBodies = append(reqBodies, reqBody)
+		}
+
+		// Dest opensearch
+		for _, destNodeIP := range config.Hardware.OpenSearchNodeIps {
+			reqBody := models.FirewallRequest{
+				SourceNodeIP:               sourceNodeIP,
+				DestinationNodeIP:          destNodeIP,
+				DestinationServicePort:     "9200",
+				DestinationServiceProtocol: "tcp",
+			}
+			reqBodies = append(reqBodies, reqBody)
+		}
+
+	}
+
+	return reqBodies
+}
+
+// getRequestsForPostgresAsSource gives the requests for all the ports where postgres is source
+func getRequestsForPostgresAsSource(config models.Config) []models.FirewallRequest {
+	var reqBodies []models.FirewallRequest
+	for _, sourceNodeIP := range config.Hardware.PostgresqlNodeIps {
+		for _, destNodeIP := range config.Hardware.PostgresqlNodeIps {
+			if sourceNodeIP != destNodeIP {
```

Comment:
This logic is correct but for better Cyclomatic complexity we can write as below

if sourceNodeIP == destNodeIP {
continue
}
//rest of the logic.

Reply:
Changed the logic